hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -211,7 +200,7 @@ public HoodieMetadataPayload(Option<GenericRecord> recordOpt) {
       key = record.get(KEY_FIELD_NAME).toString();
       type = (int) record.get(SCHEMA_FIELD_NAME_TYPE);
 
-      if (type == METADATA_TYPE_FILE_LIST || type == METADATA_TYPE_PARTITION_LIST) {
+      if (type == MetadataPartitionType.FILES.getRecordType(EMPTY_STRING) || type == MetadataPartitionType.FILES.getRecordType(RECORDKEY_PARTITION_LIST)) {
```

Comment:
For better readability, Can you introduce a method getRecordType() in MetadatPartitionType enum without any key argument and pass the correct type number back.

Reply:
done